Sources/Algorithms/Consumers/RegexConsumer.swift

```diff
@@ -15,14 +15,9 @@ public struct Regex {
 public struct RegexConsumer: CollectionConsumer {
   // NOTE: existential
   let vm: Executor
-  let referenceVM: TortoiseVM
 
   public init(regex: Regex) {
-    let ast = try! parse(regex.string, .traditional)
-    let program = Compiler(ast: ast).emit()
-    self.vm = Executor(program: program)
-    let legacyProgram = try! compile(ast, options: regex.options)
-    self.referenceVM = TortoiseVM(program: legacyProgram)
+    self.vm = _compileRegex(regex.string)
```

Comment:
Can this use the public API (e.g. `firstMatch`) instead?

Reply:
Currently it is `RegexProtocol.match(in:)`. You can add a range parameter to it.

Reply:
What would I use for types? It seems like `DynamicCaptures` will let me parse the string, but will trap at runtime:

```swift
  let regex: _StringProcessing.Regex<(DynamicCaptures)>
```

```
Could not cast value of type '()' (0x7fff815b12b8) to '_StringProcessing.DynamicCaptures' (0x10f67b618).
```

Reply:
Ah, this needs the following change:
```diff
--- a/Sources/_StringProcessing/RegexDSL/Core.swift
+++ b/Sources/_StringProcessing/RegexDSL/Core.swift
@@ -85,7 +85,13 @@ extension RegexProtocol {
     guard let result = executor.execute(input: input) else {
       return nil
     }
-    return RegexMatch(range: result.range, captures: () as! Capture)
+    let convertedCapture: Capture
+    if Capture.self == DynamicCaptures.self {
+      convertedCapture = DynamicCaptures.tuple([]) as! Capture
+    } else {
+      convertedCapture = () as! Capture
+    }
+    return RegexMatch(range: result.range, captures: convertedCapture)
   }
 }
```

Reply:
Address this later sounds good too

Reply:
Scratch that, I later also get a 

```
Could not cast value of type 'Swift.Array<Swift.Substring>' (0x7fff815ac3d8) to '()' (0x7fff815b12b8).
```